File: mobile/src/main/java/com/backdoor/moove/core/consts/LED.java
package com.backdoor.moove.core.consts;

import android.content.Context;

import com.backdoor.moove.R;

/**
 * Copyright 2016 Nazar Suhovich
 * <p/>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p/>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p/>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
public class LED {

    public static final int NUM_OF_LEDS = 17;

    public static final int WHITE = 0xffffffff;
    public static final int RED = 0xfff44336;
    public static final int GREEN = 0xff4caf50;
    public static final int BLUE = 0xff2196f3;
    public static final int ORANGE = 0xffff9800;
    public static final int YELLOW = 0xffffeb3b;
    public static final int AMBER = 0xffffc107;
    public static final int PINK = 0xffe91e63;
    public static final int GREEN_LIGHT = 0xff8bc34a;
    public static final int BLUE_LIGHT = 0xff03a9f4;
    public static final int CYAN = 0xff00bcd4;
    public static final int PURPLE = 0xff9c27b0;
    public static final int LIME = 0xffcddc39;
    public static final int INDIGO = 0xff3f51b5;
    public static final int DEEP_PURPLE = 0xff673ab7;
    public static final int DEEP_ORANGE = 0xffff5722;
    public static final int TEAL = 0xff009688;

    public static int getLED(int code) {
        int color;
        switch (code) {
            case 0:
                color = LED.WHITE;
                break;
            case 1:
                color = LED.RED;
                break;
            case 2:
                color = LED.GREEN;
                break;
            case 3:
                color = LED.BLUE;
                break;
            case 4:
                color = LED.ORANGE;
                break;
            case 5:
                color = LED.YELLOW;
                break;
            case 6:
                color = LED.PINK;
                break;
            case 7:
                color = LED.GREEN_LIGHT;
                break;
            case 8:
                color = LED.BLUE_LIGHT;
                break;
            case 9:
                color = LED.PURPLE;
                break;
            case 10:
                color = LED.AMBER;
                break;
            case 11:
                color = LED.CYAN;
                break;
            case 12:
                color = LED.LIME;
                break;
            case 13:
                color = LED.INDIGO;
                break;
            case 14:
                color = LED.DEEP_ORANGE;
                break;
            case 15:
                color = LED.DEEP_PURPLE;
                break;
            case 16:
                color = LED.TEAL;
                break;
            default:
                color = LED.BLUE;
                break;
        }
        return color;
    }

    public static String getTitle(Context context, int code) {
        String color;
        switch (code) {
            case 0:
                color = context.getString(R.string.white);
                break;
            case 1:
                color = context.getString(R.string.red);
                break;
            case 2:
                color = context.getString(R.string.green);
                break;
            case 3:
                color = context.getString(R.string.blue);
                break;
            case 4:
                color = context.getString(R.string.orange);
                break;
            case 5:
                color = context.getString(R.string.yellow);
                break;
            case 6:
                color = context.getString(R.string.pink);
                break;
            case 7:
                color = context.getString(R.string.light_green);
                break;
            case 8:
                color = context.getString(R.string.light_blue);
                break;
            case 9:
                color = context.getString(R.string.purple);
                break;
            case 10:
                color = context.getString(R.string.amber);
                break;
            case 11:
                color = context.getString(R.string.cyan);
                break;
            case 12:
                color = context.getString(R.string.lime);
                break;
            case 13:
                color = context.getString(R.string.indigo);
                break;
            case 14:
                color = context.getString(R.string.deep_orange);
                break;
            case 15:
                color = context.getString(R.string.deep_purple);
                break;
            case 16:
                color = context.getString(R.string.teal);
                break;
            default:
                color = context.getString(R.string.blue);
                break;
        }
        return color;
    }
}

File: mobile/src/main/java/com/backdoor/moove/core/consts/LED.kt
package com.backdoor.moove.core.consts

import android.content.Context

import com.backdoor.moove.R

/**
 * Copyright 2016 Nazar Suhovich
 *
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
object LED {

    val NUM_OF_LEDS = 17

    val WHITE = -0x1
    val RED = -0xbbcca
    val GREEN = -0xb350b0
    val BLUE = -0xde690d
    val ORANGE = -0x6800
    val YELLOW = -0x14c5
    val AMBER = -0x3ef9
    val PINK = -0x16e19d
    val GREEN_LIGHT = -0x743cb6
    val BLUE_LIGHT = -0xfc560c
    val CYAN = -0xff432c
    val PURPLE = -0x63d850
    val LIME = -0x3223c7
    val INDIGO = -0xc0ae4b
    val DEEP_PURPLE = -0x98c549
    val DEEP_ORANGE = -0xa8de
    val TEAL = -0xff6978

    fun getLED(code: Int): Int {
        val color: Int
        when (code) {
            0 -> color = LED.WHITE
            1 -> color = LED.RED
            2 -> color = LED.GREEN
            3 -> color = LED.BLUE
            4 -> color = LED.ORANGE
            5 -> color = LED.YELLOW
            6 -> color = LED.PINK
            7 -> color = LED.GREEN_LIGHT
            8 -> color = LED.BLUE_LIGHT
            9 -> color = LED.PURPLE
            10 -> color = LED.AMBER
            11 -> color = LED.CYAN
            12 -> color = LED.LIME
            13 -> color = LED.INDIGO
            14 -> color = LED.DEEP_ORANGE
            15 -> color = LED.DEEP_PURPLE
            16 -> color = LED.TEAL
            else -> color = LED.BLUE
        }
        return color
    }

    fun getTitle(context: Context, code: Int): String {
        val color: String
        when (code) {
            0 -> color = context.getString(R.string.white)
            1 -> color = context.getString(R.string.red)
            2 -> color = context.getString(R.string.green)
            3 -> color = context.getString(R.string.blue)
            4 -> color = context.getString(R.string.orange)
            5 -> color = context.getString(R.string.yellow)
            6 -> color = context.getString(R.string.pink)
            7 -> color = context.getString(R.string.light_green)
            8 -> color = context.getString(R.string.light_blue)
            9 -> color = context.getString(R.string.purple)
            10 -> color = context.getString(R.string.amber)
            11 -> color = context.getString(R.string.cyan)
            12 -> color = context.getString(R.string.lime)
            13 -> color = context.getString(R.string.indigo)
            14 -> color = context.getString(R.string.deep_orange)
            15 -> color = context.getString(R.string.deep_purple)
            16 -> color = context.getString(R.string.teal)
            else -> color = context.getString(R.string.blue)
        }
        return color
    }
}
